From 063d214f91e93b0057416e6723de406b3fcbfe35 Mon Sep 17 00:00:00 2001
From: Andrew Purtell <apurtell@apache.org>
Date: Fri, 15 Jan 2016 17:32:43 -0800
Subject: [PATCH 030/217] Amend HBASE-14771 RpcServer#getRemoteAddress always
 returns null

This change has been reported to cause Phoenix's PhoenixIndexRpcSchedulerTest to fail
with a NPE
---
 .../org/apache/hadoop/hbase/ipc/RpcServer.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
index 8297e90..4797143 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java
@@ -2590,7 +2590,7 @@ public class RpcServer implements RpcServerInterface, ConfigurationObserver {
    */
   public static InetAddress getRemoteIp() {
     Call call = CurCall.get();
-    if (call != null && call.connection.socket != null) {
+    if (call != null && call.connection != null && call.connection.socket != null) {
       return call.connection.socket.getInetAddress();
     }
     return null;
-- 
1.7.9.5

